qiskit/quantum_info/primitives/utils.py

```diff
@@ -0,0 +1,124 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Utility functions for primitives
+"""
+
+from __future__ import annotations
+
+import sys
+
+from qiskit.circuit import ParameterExpression, QuantumCircuit
+from qiskit.extensions.quantum_initializer.initializer import Initialize
+from qiskit.quantum_info.operators import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.states import Statevector
+
+if sys.version_info >= (3, 8):
+    # pylint: disable=no-name-in-module
+    from typing import Protocol, runtime_checkable
+else:
+    from typing_extensions import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class PauliSumOp(Protocol):
+    """Protocol for opflow.PauliSumOp."""
+
+    @property
+    def primitive(self) -> SparsePauliOp:
+        """
+        The SparsePauliOp which defines the behavior of the underlying function.
+
+        Returns:
+             The primitive SparsePauliOp.
+        """
+        ...
+
+    @property
+    def coeff(self) -> complex | ParameterExpression:
+        """
+        The scalar coefficient multiplying the Operator.
+
+        Returns:
+              The coefficient.
+        """
+        ...
+
+
+def init_circuit(state: QuantumCircuit | Statevector) -> QuantumCircuit:
+    """Initialize state."""
+    if isinstance(state, QuantumCircuit):
+        return state
+    if not isinstance(state, Statevector):
+        state = Statevector(state)
+    qc = QuantumCircuit(state.num_qubits)
+    qc.append(Initialize(state), qargs=range(state.num_qubits))
+    return qc
```

Comment:
is this needed? the circuit is unimportant since we do ideal simulation.

Reply:
`circuits` property returns QuantumCircuit. So we need this conversion anywhere. But we may not need this in the constructor. (it can be moved into `circuits`)